### PR TITLE
Fix(ComponentItem): Update the notification Id check to avoid nulls.

### PR DESCRIPTION
**Pull request by psantos1113**
Fix the sonar cloud error for potential NPE.

### CODE REVIEW

File: alert-common/src/main/java/com/synopsys/integration/alert/common/message/model/ComponentItem.java [hunk 1]
```diff
@@ -128,7 +128,8 @@ public ComponentAttributeMap getItemsOfSameName() {
         private Set<Long> notificationIds = new LinkedHashSet<>();
 
         public ComponentItem build() throws AlertException {
-            if (null == componentName || null == componentValue || null == category || null == operation || (null == notificationIds && !notificationIds.isEmpty())) {
+            boolean idsMissing = Optional.ofNullable(notificationIds).map(Collection::isEmpty).orElse(false);
```
Comment (gkillough) [inline alert-common/src/main/java/com/synopsys/integration/alert/common/message/model/ComponentItem.java]: `idsMissing` will be false if notificationIds is null. This seems to imply that the notificationIds being null means they aren't missing.  I'm not a huge fan of using optionals in this way, why not just do the null check with an `&&` inline.

Comment (gkillough) [reply alert-common/src/main/java/com/synopsys/integration/alert/common/message/model/ComponentItem.java]: Was the original bug just the negation of `notificationIds.isEmpty()`?

Comment (psantos1113) [reply alert-common/src/main/java/com/synopsys/integration/alert/common/message/model/ComponentItem.java]: sonarcloud complains about it which is the reason for this PR.

Comment (gkillough) [reply alert-common/src/main/java/com/synopsys/integration/alert/common/message/model/ComponentItem.java]: Ok, I would just prefer we create the variables without the overhead of the Optional stuff.

Comment (psantos1113) [reply alert-common/src/main/java/com/synopsys/integration/alert/common/message/model/ComponentItem.java]: I agree. I'm trying one more change first.

**Pull request merged**